### DISCUSSION TITLE
libicuuc.so.69" not found

### DISCUSSION THREAD

**Discussion by leechaowen** [termux/termux-packages#7958] [Q&A]
i run pkg update  & pkg upgrade on my phone today , the R could not working . It seems it broke the dependencies. how to fix it ?  reinstall termux ? 

https://github.com/its-pointless/gcc_termux/

~ $ R
CANNOT LINK EXECUTABLE "/data/data/com.termux/files/usr/lib/R/bin/exec/R": library "libicuuc.so.69" not found: needed by main executable
~ $ pkg reinstall r-base
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
0 upgraded, 0 newly installed, 1 reinstalled, 0 to remove and 0 not upgraded.
Need to get 0 B/43.4 MB of archives.
After this operation, 0 B of additional disk space will be used.
(Reading database ... 65456 files and directories currently installed.)
Preparing to unpack .../r-base_4.1.0_aarch64.deb ...
Unpacking r-base (4.1.0) over (4.1.0) ...
Setting up r-base (4.1.0) ...
~ $ R
CANNOT LINK EXECUTABLE "/data/data/com.termux/files/usr/lib/R/bin/exec/R": library "libicuuc.so.69" not found: needed by main executable
~ $

**Comment by rrodrigueznt** [comment]
@leechaowen, I'm facing the same issue here, and I was not sure what is the right path to follow to get R working again. I'm not a tech guy myself, but I like to learn!

I concluded that we should rebuild `r-base` to use the updated `libicu`. In my case, I updated it to 70.1 weeks ago. But I'm not sure if we should understand the magic to get it done or only developers can do it. Please, @its-pointless, could you help us?

I uninstalled `r-base` and ran `apt autoremove` to remove no longer required libraries, `libicu` 70.1 among them.

When I reinstall `r-base`, `libicu` 70.1 is also automatically installed as a dependency.

Please, 1) should I be able to install `libicu` 60.9 instead of 70.1? 2) How? It seems only  `r-base` uses that library in my installation; so, no other dependency would be broken.

Any comment will be very welcome! Thanks!

**Reply by leechaowen** [reply]
> @leechaowen, I'm facing the same issue here, and I was not sure what is the right path to follow to get R working again. I'm not a tech guy myself, but I like to learn!
> 
> I concluded that we should rebuild `r-base` to use the updated `libicu`. In my case, I updated it to 70.1 weeks ago. But I'm not sure if we should understand the magic to get it done or only developers can do it. Please, @its-pointless, could you help us?
> 
> I uninstalled `r-base` and ran `apt autoremove` to remove no longer required libraries, `libicu` 70.1 among them.
> 
> When I reinstall `r-base`, `libicu` 70.1 is also automatically installed as a dependency.
> 
> Please, 1) should I be able to install `libicu` 60.9 instead of 70.1? 2) How? It seems only `r-base` uses that library in my installation; so, no other dependency would be broken.
> 
> Any comment will be very welcome! Thanks!


Please visit  https://github.com/its-pointless/gcc_termux/issues/135

follow the way compile R from source code, hope it can help you.
